aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SubscriptionProcessor.java

```diff
@@ -114,15 +113,19 @@ void startSubscriptions() {
         return Observable.<GraphQLResponse<ModelWithMetadata<T>>>create(emitter -> {
             CountDownLatch latch = new CountDownLatch(1);
             SubscriptionMethod method = subscriptionMethodFor(appSync, subscriptionType);
-            AtomicReference<Cancelable> cancelable = new AtomicReference<>(NoOpCancelable::new);
-            emitter.setCancellable(cancelable::get);
-            cancelable.set(method.subscribe(
+            Cancelable subscriptionCancelable = method.subscribe(
                 clazz,
                 token -> latch.countDown(),
                 emitter::onNext,
                 emitter::onError,
                 emitter::onComplete
-            ));
+            );
+            // When the observable is disposed, we need to call cancel() on the subscription
+            // so it can properly dispose of resources if necessary. For the AWS API plugin,
+            // this means means closing the underlying network connection.
+            emitter.setDisposable(
+                AsyncUtils.asDisposable(subscriptionCancelable)
+            );
```

Comment:
I think this probably solves a bug I've been seeing lately. Thanks!

Reply:
Yeah...it's a pesky one too.